src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java

```diff
@@ -316,7 +343,7 @@ private int executeRemoteCommand(final Session jschSession, final String command
         try {
             // If root, modify the command to set up sudo -S
             String finalCommand = null;
-            if (executeAsRoot) {
+            if (isUnix && executeAsRoot) {
```

Comment:
Use member isUnix to pass parameter is tricky, change it to a function parameter.